aws/keys/kms.go

```diff
@@ -0,0 +1,145 @@
+package keys
+
+import (
+	"crypto"
+	"crypto/ecdsa"
+	"crypto/rsa"
+	"crypto/x509"
+	"errors"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/kms"
+)
+
+// ParseDER parses the given DER-encoded bytes and results the public or private
+// key encoded.
+func parseDER(b []byte) (interface{}, error) {
+	key, err := x509.ParsePKIXPublicKey(b)
+	if err != nil {
+		if key, err = x509.ParsePKCS1PublicKey(b); err != nil {
+			return nil, errors.New("error decoding DER; bad format")
+		}
+	}
+
+	return key, nil
+}
+
+func SignAwsKms(arn string, region string, digest []byte, opts crypto.SignerOpts, kty string) ([]byte, error) {
+	//
+	// Connect to AWS and grab the public key
+	//
+	sess, err := session.NewSession(&aws.Config{
+		Region: aws.String(region)})
```

Comment:
Rather than creating the session here, can you provide the kms client as a parameter?